src/industry_cmd.cpp

```diff
@@ -2411,20 +2402,14 @@ void GenerateIndustries()
  */
 static void UpdateIndustryStatistics(Industry *i)
 {
-	for (byte j = 0; j < lengthof(i->produced_cargo); j++) {
-		if (IsValidCargoID(i->produced_cargo[j])) {
-			byte pct = 0;
-			if (i->this_month_production[j] != 0) {
-				i->last_prod_year = TimerGameCalendar::year;
-				pct = ClampTo<byte>(i->this_month_transported[j] * 256 / i->this_month_production[j]);
-			}
-			i->last_month_pct_transported[j] = pct;
-
-			i->last_month_production[j] = i->this_month_production[j];
-			i->this_month_production[j] = 0;
+	for (auto &p : i->produced) {
+		if (IsValidCargoID(p.cargo)) {
+			if (p.history[0].PctTransported() != 0) i->last_prod_year = TimerGameCalendar::year;
```

Comment:
Does it really need to calculate a percentage here? Isn't `.transported != 0` equally good?
Well, except if you transport minute amounts of a huge production.
Furthermore, `last_prod_year` feels a bit weird when it's actually about there having been transport instead of production.

Reply:
Indeed, I was not really paying attention to what it was doing with it. I guess last_prod_year could be renamed independently at some point.

Reply:
No, I messed up completely and the original test is for production anyway.